### PR TITLE
Feat/deliverable3

### PR DESCRIPTION
**Pull request by JCoronaWize**
## Description (THIS IS A DRAFT)

- Update for Deliverable 3 of bootcamp

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking refactor which doesn't change functionality)
- [x] Styles (Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc))

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works
- [x] I have provided screenshots or recordings of results
- [x] I have added unit tests for the piece of work

# Evaluation Criteria
- [x] Navigation to all pages (Home Page, Product Page, Product Detail Page, Search Results Page) works correctly using React Router (20 points)
- [x] Featured Banners Slider, Product Categories Carousel/Grid, and Featured Products Grid on the Home Page are working correctly fetching data from the API (5 points)
- [x] Product Category Sidebar on the Product List Page is filtering Products Grid correctly interacting with the API (25 points)
- [x] Products List Grid is fetching data from the API and Pagination Controls are working correctly (15 points)
- [x] Product Detail Page is fetching data correctly from the API for the selected product and all the required blocks are working properly (15 points)
- [x] Search Results Page is connected to the search input in the header and is consuming the API to filter only the results that match the search term value from the query param (10 points)
- [x] Pagination Controls and Custom Message in Search Results Page are generated dynamically and are working correctly (10 points)
## Bonus
- [ ] React Context and useReducer or Redux are implemented to manage the global state of your application (15 points)
- [x] Abort Controller is implemented correctly for all your fetch requests (15 points)
- [x] Loading indicator appears on all the components that consume an API (10 points)
- [ ] PropTypes are implemented correctly for all your components (10 points)

# Notes 
 - With new information being added, some elements are changing in proportion and design in order to be able to show the information required.

### CODE REVIEW

File: src/Header.js [hunk 1]
```diff
@@ -8,30 +8,46 @@ import {
...
 
-const Header = ({current, onNavClick}) => {
+const Header = () => {
+  const locationQuery = useLocation().search;
+  const navigate = useNavigate();
+  const bsearch = new URLSearchParams(decodeURIComponent(locationQuery)).get(
+    "q"
+  );
+  const [searchTerm, setSearchTerm] = useState(bsearch ? bsearch : "");
+  useEffect(() => {}, [searchTerm]);
```
Comment (AlfredoGJ) [inline src/Header.js:23]: Is this `useEffect` necessary ?🤔 
```suggestion

```

File: src/pages/ProductList.js [hunk 2]
```diff
@@ -81,70 +84,126 @@ export const StyledLoader = styled.div`
   }
 `;
```
Comment (AlfredoGJ) [inline src/pages/ProductList.js:85]: Let's move all this to its own `ProductList.styled.jsx`

File: src/pages/ProductList.js [hunk 3]
```diff
@@ -81,70 +84,126 @@ export const StyledLoader = styled.div`
...
+          {productsLoading && <div>...Loading Products</div>}
+          {!productsLoading && !productsError && (
+            <>
+              <GridContainer
+                pagination={12}
+                productInfo={productsData}
+                theTitle="Product List"
+              ></GridContainer>
+            </>
+          )}
```
Comment (AlfredoGJ) [inline src/pages/ProductList.js:206]: Nice code reuse 👍

**Review by AlfredoGJ — APPROVED** [review]
## Evaluation Criteria
- [x] Navigation to all pages (Home Page, Product Page, Product Detail Page, Search Results Page) works correctly using React Router (20 points)
- [x] Featured Banners Slider, Product Categories Carousel/Grid, and Featured Products Grid on the Home Page are working correctly fetching data from the API (5 points)
- [x] Product Category Sidebar on the Product List Page is filtering Products Grid correctly interacting with the API (25 points)
- [x] Products List Grid is fetching data from the API and Pagination Controls are working correctly (15 points)
 - [x] Product Detail Page is fetching data correctly from the API for the selected product and all the required blocks are working properly (15 points)
- [x] Search Results Page is connected to the search input in the header and is consuming the API to filter only the results that match the search term value from the query param (10 points)
- [x] Pagination Controls and Custom Message in Search Results Page are generated dynamically and are working correctly (10 points)
## Bonus
- [ ] React Context and useReducer or Redux are implemented to manage the global state of your application (15 points)
- [x] Abort Controller is implemented correctly for all your fetch requests (15 points)
- [x] Loading indicator appears on all the components that consume an API (10 points)
- [ ] PropTypes are implemented correctly for all your components (10 points)

## Comments
Excellent Job!!! 💯  ✅  - All the requirements are completed plus some bonus. You are keeping the code clean, reusing, and applying really well the topics of the bootcamp. Keep it up like that 😃

**Comment by JCoronaWize** [comment]
Thank you, I'll take the observations into account for the next deliverable

**Pull request merged**